pkg/bgpv1/agent/controller.go

```diff
@@ -85,6 +93,8 @@ type ControllerParams struct {
 	Sig                     *signaler.BGPCPSignaler
 	RouteMgr                BGPRouterManager
 	PolicyResource          resource.Resource[*v2alpha1api.CiliumBGPPeeringPolicy]
+	SecretResource          resource.Resource[*slimcorev1.Secret]
+	SecretStore             diffstore.DiffStore[*slimcorev1.Secret]
```

Comment:
I don't see any use of the `Diff()` method of the `DiffStore` in the code (what is the main reason for using the `DiffStore`). In that case you can use [BGPCPResourceStore](https://github.com/cilium/cilium/blob/1e41011ee983ceefe19a0c6080be7646faaa74e2/pkg/bgpv1/manager/resource_store.go#L18-L22) instead to save some bookkeeping.

Reply:
Also, why do we need the `SecretStore` here in the `Controller`? I don't see any use of it outside of reconcilers.

Reply:
I think that's also a pre-diffstore leftover, it was there because otherwise `.Store()` on Resource would block as it wasn't called from a Start hook. Except I didn't realise it wasn't needed after switching to DiffStore. I should be able to switch to `BGPCPResourceStore` as well, I just hadn't realised that was available.

Reply:
Now updated to use BGPCPResourceStore and reverted the diffstore changes that weren't needed, so it's a bit cleaner :)

Reply:
Thanks a lot, looks much cleaner now! Left one last cleanup nit comment in https://github.com/cilium/cilium/pull/23759/files#r1345298515